src/main/java/de/hpi/bpt/argos/api/entityType/EntityTypeEndpointImpl.java

```diff
@@ -0,0 +1,240 @@
+            halt(HttpStatusCodes.ERROR, "could not read attributes.");
+        }
+        JsonArray jsonTypeAttributes = new JsonArray();
+        for (TypeAttribute att : typeAttributes) {
+            JsonObject jsonAttribute = new JsonObject();
+            jsonAttribute.addProperty("Id", att.getId());
+            jsonAttribute.addProperty("Name", att.getName());
+            jsonTypeAttributes.add(jsonAttribute);
+        }
+
+        response.body(serializer.toJson(jsonTypeAttributes));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getEntityTypeEntityMappings(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long entityTypeId = getEntityTypeId(request);
+        List<EventEntityMapping> mappings = PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEntityType(entityTypeId);
+        JsonArray jsonMappings = getMappingsJson(mappings);
+
+        response.body(serializer.toJson(jsonMappings));
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * This method returns a json containing all hierarchy objects of the given tree.
+     * @param tree map containing all hierarchy layers
+     * @param root root of the tree
+     * @return json containing all hierarchy objects
+     */
+    private JsonObject getHierarchyJson(Map<Long, List<EntityType>> tree, EntityType root) {
+        JsonArray hierarchyJson = new JsonArray();
+
+        JsonObject jsonRoot = getEntityTypeJson(root);
+        List<JsonObject> rootList = new ArrayList<>();
+        rootList.add(jsonRoot);
+
+        Map<Integer, List<JsonObject>> layerMap = new HashMap<>();
+        layerMap.put(0, rootList);
+
+        getChildJsons(tree.get(root.getId()), layerMap, 1, tree);
+        for (int layerId = 0; layerId <= layerMap.size(); layerId++) {
+            JsonArray jsonHierarchyLayer = new JsonArray();
+            for (JsonObject jsonEntityType : layerMap.get(layerId)) {
+                jsonHierarchyLayer.add(jsonEntityType);
+            }
+            hierarchyJson.add(jsonHierarchyLayer);
+        }
+
+        return hierarchyJson.getAsJsonObject();
+    }
+
+    /**
+     * This method computes the jsons for all entity types given by entityTypesOfLayer and saves them into the given map.
+     * @param entityTypesOfLayer entity types of the processed layer
+     * @param layerMap the map the entity type jsons are saved in
+     * @param layerId the hierarchy level of the entities to be processed
+     * @param tree the tree containing all entity types of this hierarchy
+     */
+    private void getChildJsons(List<EntityType> entityTypesOfLayer, Map<Integer, List<JsonObject>> layerMap, int layerId, Map<Long, List<EntityType>> tree) {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 157 characters long line (which is greater than 150 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00103)